staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -992,7 +992,21 @@ func (e *Store) finalizeDelete(obj runtime.Object, runHooks bool) (runtime.Objec
 		}
 		return obj, nil
 	}
-	return &metav1.Status{Status: metav1.StatusSuccess}, nil
+	// Return information about the deleted object, which enables clients to
+	// verify that the object was actually deleted and not waiting for finalizers.
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		return nil, err
+	}
+	gvk := obj.GetObjectKind().GroupVersionKind()
```

Comment:
ok I see now that we set Kind to resource at another place: https://github.com/kubernetes/kubernetes/blob/6defc30337365cd0b151ab5817f6f91a0d8fa407/staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go#L102.
That is definitely wrong. We should add another Resource field and deprecate kind. Should at least add a comment in API docs saying value of Kind field is resource.

Updated the code to set Kind to store.QualifiedResource.Resource.